subprojects/core-api/src/main/java/org/gradle/api/artifacts/ComponentSelection.java

```diff
@@ -34,6 +36,31 @@
      */
     ModuleComponentIdentifier getCandidate();
 
+    /**
+     * Gets the metadata of the component.
+     * The metadata may not be available, in which case {@code null} is returned.
```

Comment:
Under what circumstances is the metadata `null`? We should document this.

Reply:
Added some information, but not 100% confident it is enough. However, looking at impl and tests, I could not find extra info on the topic. So if you know better, I am all ears 😉